src/main/java/com/qubit/solution/fenixedu/integration/cgd/domain/configuration/CgdIntegrationConfiguration.java

```diff
@@ -109,4 +114,18 @@ public CgdAddressProofGenerator getAddressProofGenerator() {
         return generator;
     }
 
+    public <T extends CgdIESCodeProviderStrategyClass> T getIESCodeProvider() {
```

Comment:
@miguel-morais the code for IESCodeProvider and AddressProofGenerator is pretty similar, since both want to instantiate a class from a classname. You can refactor the code into a single method and both this methods delegate to.